.changeset/cold-snakes-train.md

```diff
@@ -0,0 +1,42 @@
+---
+"@astrojs/markdown-remark": minor
+---
+
+Adds a `data-language` attribute on the rendered `pre` elements to expose the highlighted syntax language.
+
+For example, the following markdown fenced-codeblock will expose `data-language="python"`:
+```
+\```python
+def func():
+    print('Hello Astro!')
+\```
+```
+
+This allows retrieving the language in a rehype plugin:
+```js
+// myRehypePre.js
+import { visit } from "unist-util-visit";
+export default function myRehypePre() {
+  return (tree) => {
+    visit(tree, { tagName: "pre" }, (node) => {
+      const lang = node.properties.dataLanguage;
+      [...]
+    });
+  };
+}
+```
+
+Note: if using the built-in `<Code>` component, the output of the component being flattened html, the replacement `<pre>` cannot be accessed using `{ tagName: "pre" }`.
```

Comment:
```suggestion
Note: The `<pre>` element is not exposed when using Astro's `<Code />` component which outputs flattened HTML.
```
I think we could keep this shorter here, especially since I called attention to `{ tagName: "pre" }` before the code snippet!

"outputs"? "produces"? "renders"?